gemstone/generator/generator.py

```diff
@@ -43,23 +65,39 @@ def add_ports(self, **kwargs):
         for name, T in kwargs.items():
             self.add_port(name, T)
 
+    def __hash_wire(self, connection):
```

Comment:
This can be a helper function. Shouldn't be a member function. Move to top of the file, and keep single underscore.